### PR TITLE
Add an overload for `startTransaction` that sets the created transaction to the Scope

### LINKED ISSUE #1252: Add an overload for `startTransaction` that sets the created transaction to the Scope
Today transactions need to be set to the Scope manually if not using automatic instrumentation, ideally there's an overload to overwrite this behaviour like:

`startTransaction("tr", false);`

but by default, we bound the transaction to the Scope so users don't need to do it manually.
requires changing the docs and migration page.

cc @Tyrrrz @brustolin

### PR DESCRIPTION
**Pull request by maciejwalkowiak**
## :scroll: Description

Add an overload for `startTransaction` that sets the created transaction to the Scope

## :bulb: Motivation and Context

Fixes #1252

## :green_heart: How did you test it?

Unit tests.

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [ ] I updated the docs if needed
- [ ] No breaking changes

Breaking change: methods that previously did not set transaction on the scope, now do.

### CODE REVIEW

**Comment by maciejwalkowiak** [comment]
The number of `startTransaction` methods is getting dangerously long. We provided overloaded methods to simplify users life, but now with 10 methods the getting started experience may be a bit confusing. I think we should consider having separate method name for bound transactions or passing a builder-like method parameter where all the combinations could be set.

**Review by marandaneto — APPROVED** [review]
left a comment about the param naming, other than that, LGTM

File: sentry/src/main/java/io/sentry/Hub.java [hunk 1]
```diff
@@ -572,16 +572,11 @@ public void flush(long timeoutMillis) {
...
-      final @NotNull TransactionContext transactionContext) {
-    return this.startTransaction(transactionContext, null);
-  }
-
   @Override
   public @NotNull ITransaction startTransaction(
       final @NotNull TransactionContext transactionContext,
-      final @Nullable CustomSamplingContext customSamplingContext) {
+      final @Nullable CustomSamplingContext customSamplingContext,
+      final boolean setOnScope) {
```
[the remark this answers is not in the document]
Comment (maciejwalkowiak) [reply sentry/src/main/java/io/sentry/Hub.java]: `bindToScope` sounds good. Thanks for suggestion!

**Comment by marandaneto** [comment]
@maciejwalkowiak too late but I guess we could have migrated the Android bits too, using `bindToScope` instead of doing it manually right

**Pull request merged**